packages/components/src/internal/components/search/FilterExpressionView.tsx

```diff
@@ -0,0 +1,245 @@
+                }
+            }
+        }
+    }, [field]); // leave fieldFilter out of deps list, fieldFilter is used to init once
+
+    const updateFilter = useCallback(
+        (newFilterType: any[], newFilterValue?: any, isSecondValue?: boolean, clearBothValues?: boolean) => {
+            if (!newFilterType) {
+                onFieldFilterUpdate(undefined); // remove filter
+                return;
+            }
+
+            const filterType = resolveFilterType(newFilterType?.['value'], field);
+            if (!filterType) return;
+
+            let filter: Filter.IFilter;
+
+            if (!newFilterType['valueRequired']) {
+                filter = Filter.create(resolveFieldKey(field.name, field), null, filterType);
+            } else {
+                let value = newFilterValue;
+                if (newFilterType?.['betweenOperator']) {
+                    if (clearBothValues) {
+                        value = null;
+                    } else if (isSecondValue) {
+                        value = (firstFilterValue ? firstFilterValue + ',' : '') + newFilterValue;
+                    } else {
+                        value = newFilterValue + (secondFilterValue ? ',' + secondFilterValue : '');
+                    }
+                } else if (!value && field.jsonType === 'boolean') value = 'false';
+
+                filter = Filter.create(resolveFieldKey(field.name, field), value, filterType);
+            }
+
+            onFieldFilterUpdate(filter);
+        },
+        [field, firstFilterValue, secondFilterValue]
+    );
+
+    const onFieldFilterTypeChange = useCallback(
+        (fieldname: any, filterUrlSuffix: any) => {
+            const activeFilterType = fieldFilterOptions?.find(option => option.value === filterUrlSuffix);
+            setActiveFilterType(activeFilterType);
+            setFirstFilterValue(undefined);
+            setSecondFilterValue(undefined);
+            updateFilter(activeFilterType, undefined, undefined, true);
+        },
+        [fieldFilterOptions]
+    );
+
+    const updateBooleanFilterFieldValue = useCallback(
+        (event: any) => {
+            const newValue = event.target.value;
+            setFirstFilterValue(newValue); // boolean columns don't support between operators
+            updateFilter(activeFilterType, newValue, false);
+        },
+        [activeFilterType, firstFilterValue, secondFilterValue]
+    );
+
+    const updateTextFilterFieldValue = useCallback(
+        (event: any, isNumberInput?: boolean) => {
+            const newValue = isNumberInput ? event.target.valueAsNumber : event.target.value;
+            const isSecondInput = event.target.name.endsWith('-second');
+            if (isSecondInput) setSecondFilterValue(newValue);
+            else setFirstFilterValue(newValue);
+            updateFilter(activeFilterType, newValue, isSecondInput);
+        },
+        [activeFilterType, firstFilterValue, secondFilterValue]
```

Comment:
I initially added them here because though this function doesn't have direct dep on them, it does have indirect dependency through updateFilter call. I had to add them here explicitly for updateFilter to do the right thing. 

This type of transitive dependency is too messy to deal with, so I end up refactoring the functions to get rid of such deps.

Reply:
Update: I had to rollback the changes. When I tried removing indirect deps by passing those hook values directly as function parameters, the react component fails to update. I added comment why those extra deps are needed.